falcon/errors.py

```diff
@@ -190,9 +223,10 @@ class HTTPBadRequest(HTTPError):
     (See also: RFC 7231, Section 6.5.1)
 
     Note:
-        All the arguments should be passed as keyword only. Using them as positional
-        arguments will raise a deprecation warning and will result in an
-        error in a future version of falcon.
+        All the arguments are defined as keyword-only.
+
+        Attempting to pass them as positional arguments will make
+        the interpreter raise an instance of :class:`TypeError`.
```

Comment:
I don't think we need this note. Seems a cause a bit too much noise in the docs. 

The docs have `Parameters:` or `Keyword Arguments:`, I think it's clear enough

Reply:
OK, as discussed on Gitter, let's go for a single line statement without the preceding `NOTE:`.